kv/index.go

```diff
@@ -277,6 +277,36 @@ func (i *Index) Walk(ctx context.Context, tx Tx, foreignKey []byte, visitFn Visi
 	return indexWalk(ctx, cursor, sourceBucket, visitFn)
 }
 
+// First returns the first entry in the index.
+func (i *Index) First(tx Tx, foreignKey []byte) ([]byte, error) {
+	// return not found if the index is being used purely to write.
+	if !i.canRead {
+		return nil, ErrKeyNotFound
```

Comment:
❤️

Reply:
removed `First` because we do not need that anymore